mcp/shared.go

```diff
@@ -132,7 +133,8 @@ func handleReceive[S Session](ctx context.Context, session S, jreq *jsonrpc.Requ
 	}
 
 	mh := session.receivingMethodHandler().(MethodHandler)
-	req := info.newRequest(session, params)
+	ti, _ := jreq.Meta.(*auth.TokenInfo)
```

Comment:
I don't think we should use auth.TokenInfo as the meta value. I think it should be something like mcp.MessageMeta or mcp.MessageExtra, which has AuthInfo as a field (this is almost exactly what typescript does).

The reason is that we're going to say "Transports may optionally set req.Meta to a mcp.MessageMeta value to pass addtional information to the application, such as auth info.".

If we're choosing this API, we need to leave it open to hold more than just auth info in the future.

Reply:
RequestMeta? ServerRequestMeta?
(both confusing, actually, since there is a Params.Meta already)
So maybe (Server)?RequestExtra.